traitsui/testing/tester/registry.py

```diff
@@ -215,12 +231,60 @@ def get_handler(self, target_class, interaction_class):
         handler : callable(UIWrapper, interaction) -> any
             The function to handle the particular interaction on a target.
             ``interaction`` should be an instance of ``interaction_class``.
+
+        Raises
+        ------
+        InteractionNotSupported
+            If the given target and interaction types are not supported
+            by this registry.
         """
         return self._interaction_registry.get_value(
             target_class=target_class,
             key=interaction_class,
         )
 
+    def get_interactions(self, target_class):
+        """ Returns all the interactions supported for the given target type.
+
+        Parameters
+        ----------
+        target_class : subclass of type
+            The type of a UI target being operated on.
+
+        Returns
+        -------
+        interaction_classes : set
+            Supported interaction types for the given target type.
+        """
+        return self._interaction_registry.get_keys(target_class=target_class)
+
+    def get_interaction_doc(self, target_class, interaction_class):
+        """ Return the documentation for the given target and locator type.
+
+        Parameters
+        ----------
+        target_class : subclass of type
+            The type of a UI target being operated on.
+        interaction_class : subclass of type
+            Any class.
+
+        Returns
+        -------
+        doc : str
+
+        Raises
+        ------
+        InteractionNotSupported
+            If the given target and interaction types are not supported
+            by this registry.
+        """
+        self._interaction_registry.get_value(
+            target_class=target_class,
+            key=interaction_class,
+        )
+        # This maybe configurable in the future via register_handler
```

Comment:
I was thinking of adding an optional argument `help` to `register_handler` so that implementations can modify the help message with more context specific documentation. e.g. For the slider in RangeEditor, we can document the fact that the `KeyClick` command only supports "Up", "Down", ... keys. But that would be a nice-to-have. Getting the documentation from the class is not so bad - the generic description is already quite useful.

Similarly for `register_solver` for registering location solver.
(I can remove these comments too if the reviewer finds them out-of-place.)